### PR TITLE
GDLint + flycheck support added

### PR DESCRIPTION
**Pull request by ofrank123**
I've added support for gdlint with flycheck.  Unfortunately, this means requiring flycheck as a dependency, but it is not possible to avoid this without adding the gdlint checker to flycheck itself, as the compiler fails if I try to use something like `eval-on-load` or a hook.

### CODE REVIEW

**Comment by NathanLovato** [comment]
I've got to see with the melpa team, as they have lots of `flycheck-language` packages, and their contribution guidelines state that we should provide separate packages for third-party features. If it requires flycheck, that an emacs user may not have installed, this won't pass review I think.

We could have this in a small package in a separate repository in that case. I'll just have to check.

**Comment by NathanLovato** [comment]
Could you please follow our commit guidelines? Don't use passive, start with a verb, and write a sentence in plain English. https://www.gdquest.com/docs/guidelines/contributing-to/gdquest-projects/#commit-guidelines

In this case, for example, `Add linter support with gdlint and flycheck`

**Comment by NathanLovato** [comment]
Also it looks good, just let me check with the melpa people before merging, we might need to have this in a different repo.

**Comment by ofrank123** [comment]
Yeah, I think it may be better to make a PR on the flycheck repo. I'll look into this and fix the commit message in a bit, sorry about that.

**Comment by NathanLovato** [comment]
> Yeah, I think it may be better to make a PR on the flycheck repo.

Actually, flycheck being a base framework, flycheck checkers are all individual packages. I'll open a new repository here for this and we can submit it to melpa later.

**Comment by ofrank123** [comment]
We can just leave this PR open until that gets sorted then.

**Comment by NathanLovato** [comment]
I opened a new repository for this, could you commit your work there? https://github.com/GDQuest/emacs-gdscript-flycheck
You should have write access. Thanks for your work!

**Comment by NathanLovato** [comment]
Could you rename the package and file to `flycheck-gdscript`? That's the convention for flycheck extensions in MELPA
